tools/build/Library/ReleaseCreator.php

```diff
@@ -370,7 +370,7 @@ protected function setConfigDefinesConstants()
      */
     protected function getCurrentVersion()
     {
-        $kernelFile = $this->projectPath.'/app/AppKernel.php';
+        $kernelFile = $this->projectPath.'/src/Core/Version.php';
```

Comment:
Nice idea from @amaury-hanser 
Maybe, to not disturb generation of nightlies for the previous versions we can search in class Version and if $matches is empty, search in AppKernel